R/coord-.r

```diff
@@ -71,13 +81,25 @@ Coord <- ggproto("Coord",
     return(list(x = scale_details$x.range, y = scale_details$y.range))
   },
 
-  train = function(scale_details) NULL,
+  train = function(scale_details, params = list()) NULL,
 
   transform = function(data, range) NULL,
 
   distance = function(x, y, scale_details) NULL,
 
-  is_linear = function() FALSE
+  is_linear = function() FALSE,
+
+  setup_params = function(data) {
```

Comment:
I don't think this - this works like the `setup_params` for position, not like the other classes. It's all a bit inconsistent, but I can't see a way to do better without a huge amount of work.